unix/client/dns_client_unix.ml

```diff
@@ -35,23 +36,34 @@ module Uflow : Dns_client_flow.S
     end >>= fun proto_number ->
     let socket = Unix.socket PF_INET SOCK_STREAM proto_number in
```

Comment:
- `Unix.socket` can also fail, we should probably guard that too. Technically speaking I think the same goes for `getprotobyname` which I guess can fail if e.g. the application is running from within a chroot or sandbox that doesn't have access to `/etc/protocols`.
  - Same for the equivalent `Lwt` wrappers in `lwt/client/dns_client_lwt.ml`

Reply:
well spotted, I implemented these guards in 14c6859